lib/init.c

```diff
@@ -84,6 +84,19 @@
 
 
 
+#if defined(__amigaos4__) || defined(__AMIGA__) || defined(__AROS__)
+#include <errno.h>
+#define getlogin_r(a,b) ENXIO
+#ifndef __AROS__
+#define srandom srand
+#define random rand
+#endif
+#ifndef __amigaos4__
+#include <proto/bsdsocket.h>
+#define close CloseSocket
+#endif
+#endif // __amigaos4__
+
```

Comment:
Lets move this into compat.h

Reply:
Moved here:
https://github.com/sahlberg/libsmb2/pull/297/commits/99944b39e6ceebf7603c3dd189222a666a8643aa